File: src/main/java/de/cplaiz/activecraftcore/commands/WhereAmICommand.java
package de.cplaiz.activecraftcore.commands;

import de.cplaiz.activecraftcore.ActiveCraftPlugin;
import de.cplaiz.activecraftcore.exceptions.ActiveCraftException;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.List;

public class WhereAmICommand extends ActiveCraftCommand {

    public WhereAmICommand(ActiveCraftPlugin plugin) {
        super("whereami",  plugin);
    }

    @Override
    public void runCommand(CommandSender sender, Command command, String label, String[] args) throws ActiveCraftException {
        CommandTargetType type = args.length >= 1 ? CommandTargetType.OTHERS : CommandTargetType.SELF;
        Player target = type == CommandTargetType.SELF ? getPlayer(sender) : getPlayer(args[0]);
        checkPermission(sender, type.code());
        if (args.length != 0)
            isTargetSelf(sender, target);
        String coords = ChatColor.GOLD + "x" + ChatColor.AQUA + target.getLocation().getBlockX() + ChatColor.GOLD
                + " y" + ChatColor.AQUA + target.getLocation().getBlockY() + ChatColor.GOLD +
                " z" + ChatColor.AQUA + target.getLocation().getBlockZ();
        messageFormatter.setTarget(getProfile(target));
        messageFormatter.addReplacements("world", target.getWorld().getName(), "coords", coords);
        sendMessage(sender, cmdMsg(type.code()));
    }

    @Override
    public List<String> onTab(CommandSender sender, Command command, String label, String[] args) {
        return args.length == 1 ? getBukkitPlayernames() : null;
    }

}

File: src/main/java/org/activecraft/activecraftcore/commands/WhereAmICommand.kt
package org.activecraft.activecraftcore.commands

import org.activecraft.activecraftcore.ActiveCraftPlugin
import org.activecraft.activecraftcore.exceptions.ActiveCraftException
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandSender

class WhereAmICommand(plugin: ActiveCraftPlugin?) : ActiveCraftCommand("whereami", plugin!!) {
    @Throws(ActiveCraftException::class)
    public override fun runCommand(sender: CommandSender, command: Command, label: String, args: Array<String>) {
        val type = if (args.isNotEmpty()) CommandTargetType.OTHERS else CommandTargetType.SELF
        val target = if (type == CommandTargetType.SELF) getPlayer(sender) else getPlayer(args[0])
        assertCommandPermission(sender, type.code())
        if (args.isNotEmpty()) isTargetSelf(sender, target)
        val coords = (ChatColor.GOLD.toString() + "x" + ChatColor.AQUA + target.location.blockX + ChatColor.GOLD
                + " y" + ChatColor.AQUA + target.location.blockY + ChatColor.GOLD +
                " z" + ChatColor.AQUA + target.location.blockZ)
        messageFormatter.setTarget(getProfile(target))
        messageFormatter.addFormatterPatterns("world" to target.world.name, "coords" to coords)
        sendMessage(sender, cmdMsg(type.code()))
    }

    public override fun onTab(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ) = if (args.size == 1) getBukkitPlayernames() else null

}